### PR TITLE
Auto notify tuleap about current build status

### PR DESCRIPTION
**Pull request by gorkat**
part of: [story #18320](https://tuleap.net/plugins/tracker/?aid=18320) native support of pull requests in tuleap branch source jenkins plugin

## How to test

1. Go to a repository of your choice on your Tuleap instance
2. make a pullrequest containing a Jenkinsfile like this:

```
pipeline {
    agent any

    stages {
        stage('Build') {
            steps {
                echo 'Building..'
                sh "sleep 10s"
            }
        }
        stage('Test') {
            steps {
                echo 'Testing..'
            }
        }
        stage('Deploy') {
            steps {
                echo 'Deploying....'
            }
        }
    }
}
```
3. Open the pullrequest UI -> CI status: Unknown

4. Go to your repository on Jenkins and click the [Scan multibranch
   pipeline now]

5. Once the build is started, refresh the PR UI. You should see CI
   status: Pending

6. Once the build is complete, refresh the UI once again -> CI Status:
   success

### CODE REVIEW

**Review by Erwyn — CHANGES_REQUESTED** [review]

**Comment by Erwyn** [comment]
You will need to put tuleap-api to version `2.2.0` in the `pom.xml` and push it here. I can't do it for you as it is on your forked repository sadly.

**Review by Erwyn — CHANGES_REQUESTED** [review]
You will need to put tuleap-api to version `2.2.0` in the `pom.xml` and push it here. I sadly can't do it for you as it is on your forked repository.

File: src/main/java/org/jenkinsci/plugins/tuleap_git_branch_source/TuleapSCMSource.java [hunk 1]
```diff
@@ -277,6 +278,19 @@ public String getGitBaseUri() {
         return TuleapConfiguration.get().getGitBaseUrl();
     }
 
+    public TuleapGitRepository getTuleapGitRepository() {
+        return this.repository;
+    }
+
+    @Nullable
+    public TuleapAccessToken getCredentials() {
```
[the remark this answers is not in the document]
Comment (gorkat) [reply src/main/java/org/jenkinsci/plugins/tuleap_git_branch_source/TuleapSCMSource.java:286]: I agree, I'll see what can be done for this point :)

[the remark this answers is not in the document]
Comment (gorkat) [reply src/main/java/org/jenkinsci/plugins/tuleap_git_branch_source/TuleapSCMSource.java]: done

**Pull request merged**